.github/workflows/codacy-analysis.yml

```diff
@@ -0,0 +1,30 @@
+name: Codacy Analysis
+
+on:
```

Comment:
I believe these jobs/steps need to be limited to only run on pull_request from this same repo, not forks, as PRs from forks wouldn't have access to the project token secret